steamworks-sys/build.rs

```diff
@@ -1,62 +1,48 @@
+//! Generates bindings for steamworks sdk (1.61)
+//!
+//! The steamworks dynamic library is copied to the target directory upon invoking this build script. A
+//! user is expected to load the dynamic library at runtime using `libloading` crate.
+
 #[cfg(feature = "rebuild-bindings")]
 extern crate bindgen;
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    use std::env;
-    use std::fs::{self};
-    use std::path::{Path, PathBuf};
+    let out_dir = std::env::var("OUT_DIR").unwrap();
 
-    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
-    let sdk_loc = if let Ok(sdk_loc) = env::var("STEAM_SDK_LOCATION") {
-        Path::new(&sdk_loc).to_path_buf()
-    } else {
-        let mut path = PathBuf::new();
-        path.push(env::var("CARGO_MANIFEST_DIR").unwrap());
-        path.push("lib");
-        path.push("steam");
-        path
-    };
-    println!("cargo:rerun-if-env-changed=STEAM_SDK_LOCATION");
+    // https://github.com/rust-lang/cargo/issues/9661
+    // Typically points to: $PROJECT/$TARGET_DIR/(x86_64-pc-windows-msvc/)$PROFILE/build/project-495fd7493ea5486b/out
+    // CARGO_(BUILD_)TARGET_DIR is only set by the `cargo` user, `cargo` itself **DOES NOT** set it
+    // when the user passes --target-dir: https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir
+    let target_dir = std::path::Path::new(&out_dir).join("../../..");
 
-    let triple = env::var("TARGET").unwrap();
-    let mut lib = "steam_api";
-    let mut link_path = sdk_loc.join("redistributable_bin");
-    if triple.contains("windows") {
-        if !triple.contains("i686") {
-            lib = "steam_api64";
-            link_path.push("win64");
-        }
-    } else if triple.contains("linux") {
-        if triple.contains("i686") {
-            link_path.push("linux32");
+    let sdk_src = std::path::Path::new(&std::env::var("CARGO_MANIFEST_DIR").unwrap())
+        .join("lib")
+        .join("steam");
+
+    let triple = std::env::var("TARGET").unwrap();
+
+    let dylib_src = sdk_src.join("redistributable_bin").join({
+        if triple.contains("windows") {
+            if !triple.contains("i686") {
+                // This dll has been renamed from `steam_api64` to `steam_api`.
+                "win64/steam_api.dll"
+            } else {
+                panic!("Unsupported OS");
+            }
```

Comment:
Nit: maybe better to explicitly check for `x86_64` and panic otherwise, or does `win64/steam_api.dll` support `aarch64-pc-windows-msvc` too? Want to prevent accidental issues when we inevitably bring up this platform.

Reply:
So my suggestion is basically:
```suggestion
            if triple.starts_with("x86_64") {
                // This dll has been renamed from `steam_api64` to `steam_api`.
                "win64/steam_api.dll"
            } else {
                panic!("Unsupported Windows architecture {triple{");
            }
```